src/hooks/useTypedSignTrustCredential.ts

```diff
@@ -0,0 +1,118 @@
+import type { SignTypedDataArgs } from '@wagmi/core';
+import { useCallback, useEffect, useState } from 'react';
+import {
+  useAccount,
+  useChainId,
+  usePublicClient,
+  useSignTypedData,
+} from 'wagmi';
+
+import { generateTCTypedSignPayload } from '../utils';
+
+export enum SignatureErrorTypes {
+  Invalid = 'invalid',
+  Error = 'error',
+}
+
+export type SignatureError = {
+  type: SignatureErrorTypes;
+  message: string;
+};
+
+/**
+ * A Hook to sign TrustCredential typed data.
+ *
+ * @returns The signature and the function to sign the message. isLoading and isVerified are used to indicate the status of the signature.
+ */
+export function useTypedSignTrustCredential() {
+  const client = usePublicClient();
+  const chainId = useChainId();
+  const { address } = useAccount();
+  const { data: signature, signTypedData } = useSignTypedData();
+
+  const [trustCredentialTypedData, setTrustCredentialTypedData] =
+    useState<SignTypedDataArgs>();
+  // above useSignTypedData have isLoading, but for some reason, it is not working well with loading state, use following instead
+  const [isLoading, setIsLoading] = useState(false);
+  const [isVerified, setIsVerified] = useState(false);
+  // Type [key_4: string] is defined inside the  SignTypedDataArgs in Wagmi
+  const [payload, setPayload] = useState<{ [key_4: string]: unknown }>();
+  const [signatureError, setSignatureError] = useState<SignatureError>();
+
+  const submitTypedSignRequest = useCallback(
+    (subjectAddress: string, isAdd: boolean) => {
+      if (!address) {
+        return;
+      }
+      setIsLoading(true);
```

Comment:
recomment to move the setIsLoading to useEffect?

Reply:
This `setIsLoading` need to be set in here to indicate signing process currently in progress, and `isLoading` will be set to false in useEffect code after finish verifying signature is valid and is from owner.